jax/experimental/array_api/_utility_functions.py

```diff
@@ -23,3 +28,80 @@ def all(x, /, *, axis=None, keepdims=False):
 def any(x, /, *, axis=None, keepdims=False):
   """Tests whether any input array element evaluates to True along a specified axis."""
   return jax.numpy.any(x, axis=axis, keepdims=keepdims)
+
+class __array_namespace_info__:
+
+  def __init__(self):
+    self._default_dtypes = self._build_default_dtype_dict()
+    self._capabilities = {
+      "boolean indexing": True,
+      "data-dependent shapes": False,
+    }
+    self._data_types = self._build_dtype_dict()
+
+  def _build_default_dtype_dict(self):
+    default_dtypes = {
+      "real floating": "f",
+      "complex floating": "c",
+      "integral": "i",
+      "indexing": "i",
+    }
+    for dtype_name, kind in default_dtypes.items():
+      dtype = _dtypes._default_types.get(kind)
+      dtype = _dtypes.canonicalize_dtype(dtype)
+      default_dtypes[dtype_name] = dtype
+    return default_dtypes
+
+  def _build_dtype_dict(self):
+    data_types = {
+      "signed integer": ["int8", "int16", "int32", "int64"],
+      "unsigned integer": ["uint8", "uint16", "uint32", "uint64"],
+      "real floating": ["float32", "float64"],
+      "complex floating": ["complex64", "complex128"],
+    }
+    if not config.enable_x64.value:
+      for category in data_types:
+        data_types[category] = data_types[category][:-1]
+
+    data_types["bool"] = ["bool"]
+
+    for category in data_types:
+      _dtype_dict = {}
+      for name in data_types[category]:
+        _dtype_dict[name] = _dtypes.dtype(name)
+      data_types[category] = _dtype_dict
+    data_types["integral"] = (
+      data_types["signed integer"] | data_types["unsigned integer"]
+    )
+    data_types["numeric"] = (
+      data_types["integral"]
+      | data_types["real floating"]
+      | data_types["complex floating"]
+    )
+    return data_types
+
+  def default_device(self):
+    # By default JAX arrays are uncommitted (device=None), meaning that
+    # JAX is free to choose the most efficient device placement.
+    return None
+
+  def devices(self):
+    return jax.devices()
+
+  def capabilities(self):
+    return self._capabilities
+
+  def default_dtypes(self):
```

Comment:
It looks like this should accept a keyword-only argument `device=None`: https://data-apis.org/array-api/latest/API_specification/generated/array_api.info.default_dtypes.html#array_api.info.default_dtypes